### PR TITLE
refactor: code file breakdown and reorg

### LINKED ISSUE #100: Refactor: Clean up internal file structure for entire library
Auto-Entity started out as a simple idea, and has since over the last year and a half grown into something much richer and more complex. The internal files of the library are growing unruly, and could use some cleanup. 

Refactor the library to break down large and complex code files into more smaller code files with more targeted functionality, move code files into more folders, and update the public interface as necessary.

### PR DESCRIPTION
**Pull request by jrista**
* Break actions into multiple files under /lib/actions
 * Break decorators into multiple files under /lib/decorators
 * Break effects into multiple files under /lib/effects
 * Move operators under /lib/effects
 * Move reducer under /lib/reducer
 * Break service down into multiple files under /lib/service
 * Break util down into multiple files under /lib/util
 * Update library index to properly reference all files
 * Update each area of library to properly reference files
 * Move uuid generator function to /util

Resolves #100

### CODE REVIEW

File: projects/ngrx-auto-entity/src/lib/service/error-handling.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+export const logAndThrow = (method: string, err: any, entityInfo: IEntityInfo) => {
+  console.error(`[NGRX-AE] ! Service error: ${method}(). (Entity: ${entityInfo.modelName})`);
+  console.error(err);
+  return throwError({ info: entityInfo, err });
+};
+
+export const logServiceLocateFailure = (entityInfo: IEntityInfo, serviceName: string): void =>
+  console.error(
+    `[NGRX-AE] ! Error: Unable to locate service "${serviceName}" using model name of "${entityInfo.modelName}"`
```
Comment (jrista) [inline projects/ngrx-auto-entity/src/lib/service/error-handling.ts]: This message needs to be refactored at some point. Very early versions of auto-entity tried to find the service based on the model name, however that approach was abandoned in favor of the current provider mapping approach. This message should reflect that AN entity service, any entity service, could not be found for the specified model. (This fix can happen in a future branch.)

File: projects/ngrx-auto-entity/src/lib/service/service-invocation.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export const callService = <TModel, TModelObs, TResult>(
+  method: string,
+  entityInfo: IEntityInfo,
+  injector: Injector,
+  invoke: (service: IAutoEntityService<TModel>) => Observable<TModelObs>,
+  toResult: (entity: TModelObs) => TResult
+): Observable<TResult> => {
+  try {
+    const service = getService(entityInfo, injector);
```
Comment (jrista) [inline projects/ngrx-auto-entity/src/lib/service/service-invocation.ts]: Should probably move `getService` function into this file.

File: projects/ngrx-auto-entity/src/index.ts [hunk 3]
```diff
@@ -103,62 +96,57 @@ export {
...
 
+export { ENTITY_OPTS_PROP } from './lib/decorators/entity-tokens';
+export { IEffectExcept, IEntityOptions, IEntityTransformer, Entity } from './lib/decorators/entity';
+export { IEffectExclusions } from './lib/decorators/effect-exclusions';
+export { curd } from './lib/decorators/effect-exclusion-utils';
+export { loads } from './lib/decorators/effect-exclusion-utils';
+export { extra } from './lib/decorators/effect-exclusion-utils';
+export { all } from './lib/decorators/effect-exclusion-utils';
+export { matching } from './lib/decorators/effect-exclusion-utils';
+export { except } from './lib/decorators/effect-exclusion-utils';
```
Comment (patpaquette) [inline projects/ngrx-auto-entity/src/index.ts]: Are these separate exports intended or should they be grouped?

**Review by patpaquette — APPROVED** [review]

File: projects/ngrx-auto-entity/src/index.ts (hunk 3 above)
Comment (jrista) [reply projects/ngrx-auto-entity/src/index.ts]: They should be grouped. I think this is a consequence of the extract to file process in webstorm.

**Pull request merged**